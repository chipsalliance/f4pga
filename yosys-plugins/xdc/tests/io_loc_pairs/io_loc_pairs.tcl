yosys -import
if { [info procs get_ports] == {} } { plugin -i design_introspection }
if { [info procs read_xdc] == {} } { plugin -i xdc }
yosys -import  ;# ingest plugin commands

read_verilog $::env(DESIGN_TOP).v

read_verilog -lib -specify +/xilinx/cells_sim.v
read_verilog -lib +/xilinx/cells_xtra.v
read_verilog -lib [file dirname $::env(DESIGN_TOP)]/cells_xtra.v

hierarchy -check -top top

# -flatten is used to ensure that the output eblif has only one module.
# Some of F4PGA expects eblifs with only one module.
synth_xilinx -flatten -abc9 -nosrl -noclkbuf -nodsp -run prepare:check

#Read the design constraints
read_xdc -part_json [file dirname $::env(DESIGN_TOP)]/../xc7a35tcsg324-1.json $::env(DESIGN_TOP).xdc

# Clean processes before writing JSON.
yosys proc

# Write the design in JSON format.
write_json [test_output_path "io_loc_pairs.json"]
write_blif -param [test_output_path "io_loc_pairs.eblif"]
