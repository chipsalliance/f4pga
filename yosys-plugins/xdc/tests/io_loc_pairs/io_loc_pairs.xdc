# OBUF_6
set_property LOC D10 [get_ports {led[0]}]
set_property DRIVE 12 [get_ports {led[0]}]

# OBUF_7
set_property LOC A9 [get_ports {led[1]}]
set_property IN_TERM UNTUNED_SPLIT_40 [get_ports led[1]]
set_property SLEW FAST [get_ports led[1]]
set_property IOSTANDARD SSTL135 [get_ports led[1]]

# OBUF_OUT
set_property LOC E3 [get_ports out_a]
set_property IN_TERM UNTUNED_SPLIT_50 [get_ports out_a]
set_property SLEW FAST [get_ports out_a]
set_property IOSTANDARD LVCMOS33 [get_ports out_a]

# bottom_inst.OBUF_10
set_property LOC C2 [get_ports {out_b[0]}]
set_property SLEW SLOW [get_ports {out_b[0]}]
set_property IOSTANDARD LVCMOS18 [get_ports {out_b[0]}]

# bottom_inst.OBUF_11
set_property LOC R2 [get_ports {out_b[1]}]
set_property DRIVE 4 [get_ports {out_b[1]}]
set_property SLEW FAST [get_ports {out_b[1]}]
set_property IOSTANDARD LVCMOS25 [get_ports {out_b[1]}]

# bottom_inst.OBUF_9
set_property LOC M6 [get_ports {led[2]}]
set_property SLEW FAST [get_ports {led[2]}]
set_property IOSTANDARD DIFF_SSTL135 [get_ports {led[2]}]

# bottom_intermediate_inst.OBUF_8
set_property LOC N4 [get_ports {led[3]}]
set_property DRIVE 16 [get_ports {led[3]}]
set_property IOSTANDARD SSTL135 [get_ports {led[3]}]

# OBUFTDS_2
set_property LOC N2 [get_ports signal_p]
set_property LOC N1 [get_ports signal_n]
set_property SLEW FAST [get_ports signal_p]
set_property IOSTANDARD DIFF_SSTL135 [get_ports signal_p]

# GTPE2_CHANNEL
set_property LOC G1 [get_ports {rx_p}]
set_property LOC G2 [get_ports {rx_n}]
set_property LOC G3 [get_ports {tx_p}]
set_property LOC G4 [get_ports {tx_n}]

# IBUFDS_GTE2
set_property LOC G5 [get_ports {ibufds_gte2_i}]
set_property LOC G6 [get_ports {ibufds_gte2_ib}]
