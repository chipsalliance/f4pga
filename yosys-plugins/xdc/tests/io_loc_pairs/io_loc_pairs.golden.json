{
    "GTPE2_CHANNEL": {
        "IO_LOC_PAIRS": "rx_p:G1,rx_n:G2,tx_p:G3,tx_n:G4"
    },
    "IBUFDS_GTE2": {
        "IO_LOC_PAIRS": "ibufds_gte2_i:G5,ibufds_gte2_ib:G6"
    },
    "OBUFTDS_2": {
        "IOSTANDARD": "DIFF_SSTL135",
        "IO_LOC_PAIRS": "signal_p:N2,signal_n:N1",
        "SLEW": "FAST"
    },
    "OBUF_6": {
        "DRIVE": "12",
        "IOSTANDARD": "LVCMOS33",
        "IO_LOC_PAIRS": "led[0]:D10",
        "SLEW": "SLOW"
    },
    "OBUF_7": {
        "IN_TERM": "UNTUNED_SPLIT_40",
        "IOSTANDARD": "SSTL135",
        "IO_LOC_PAIRS": "led[1]:A9",
        "SLEW": "FAST"
    },
    "OBUF_OUT": {
        "IN_TERM": "UNTUNED_SPLIT_50",
        "IOSTANDARD": "LVCMOS33",
        "IO_LOC_PAIRS": "out_a:E3",
        "SLEW": "FAST"
    },
    "bottom_inst.OBUF_10": {
        "IOSTANDARD": "LVCMOS18",
        "IO_LOC_PAIRS": "out_b[0]:C2",
        "SLEW": "SLOW"
    },
    "bottom_inst.OBUF_11": {
        "DRIVE": "4",
        "IOSTANDARD": "LVCMOS25",
        "IO_LOC_PAIRS": "out_b[1]:R2",
        "SLEW": "FAST"
    },
    "bottom_inst.OBUF_9": {
        "IOSTANDARD": "DIFF_SSTL135",
        "IO_LOC_PAIRS": "led[2]:M6",
        "SLEW": "FAST"
    },
    "bottom_intermediate_inst.OBUF_8": {
        "DRIVE": "16",
        "IOSTANDARD": "SSTL135",
        "IO_LOC_PAIRS": "led[3]:N4",
        "SLEW": "SLOW"
    }
}
